### ISSUE TITLE
[OAuth 2.0] "invalid_request" for changing outlook incoming server authentication to OAuth 2.0

### ISSUE THREAD

**Issue by Gravemind2015** [thunderbird/thunderbird-android#8016]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.904

### Where did you get the app from?

F-Droid

### Android version

OneUI 4.1 (Android 12)

### Device model

Samsung M51

### Steps to reproduce

Context: Got a microsoft email that older sign-in methods will be disabled soon. Switching to OAuth 2,0 in thunderbird desktop was easy (sending a test email triggered OAuth sign-in). Then logged in to my account settings through browser and cleared all app passwords. This caused K-9 mail on my phone to show 'Authentication failed' notification. Tapping on it led to the 'Inoming Server Settings' page of that outlook account.

Steps performed then:
1. Changed Authentication from 'Normal Password' to 'OAuth 2.0'.
2. Clicked next, it prompted me to Sign in (We'll redirect you to your email provider to sign in. You need to grant the app access to your email account).
3. Tapping on the link opened my browser (Tried a bunch: Firefox, Fennec, Chrome, Cromite, even copied URL to desktop and tried in Firefox desktop. Same error).
4. The page that opens throws an error.

### Expected behavior

The website should prompt to sign in and allow adding k-9 as an email client.

### Actual behavior

The page shows an error instead:

We're unable to complete your request

invalid_request: The provided value for the input parameter 'redirect_uri' is not valid. The expected value is a URI which matches a redirect URI registered for this client application

I have tried manually capturing the url and manually pasting (just in case) but it always errors out. Also it doesn't matter if that account is already logged into the browser, it never detects it.

### Logs

![Screenshot_20240720-164130_Firefox Focus](https://github.com/user-attachments/assets/e73795b7-358a-4d8b-b954-124abc130f41)

**Comment by xerta555** [comment]
This is going to become a real problem for K9.

Support for modern authentication is becoming mandatory.

**Comment by nICEnnnnnnnLee** [comment]
And I changed IMAP server from `imap-mail.outlook.com` to `outlook.office365.com` according to [microsoft docs](https://support.microsoft.com/en-us/office/pop-imap-and-smtp-settings-for-outlook-com-d088b986-291d-42b8-9564-9c414e2aa040).  
Don't know if it matters or not.

**Comment by Gravemind2015** [comment]
@nICEnnnnnnnLee Thanks for the hint, I downgraded back to the 6.804 (fdroid) version to test and while it still failed the first time when I re-imported my backup and tried to reauthenticate, changing my default browser to another browser somehow fixed it.

This was weird so to try again, I added a completely new account, and this time even the old browser started loading the OAuth sign-in page correctly. I guess K-9 mail was generating the correct URL but the browser (whichever was the default) was receiving a broken one and thus the 'invalid_request' error.

Unfortunately I can't reproduce it now to confirm. but it's fixed and I've successfully converted all outlook accounts to OAuth 2.0.
Thank you, I guess I should close the issue.
Regards,
Gravemind2015
